packages/prop-controllers/src/prop-controllers.ts

```diff
@@ -75,16 +76,45 @@ export type Data =
   | Data[]
   | { [key: string]: Data }
 
-export type ElementData = {
-  type: string
-  key: string
-  props: Record<string, Data>
+export const dataSchema: z.ZodType<Data> = z.union([
+  z.undefined(),
+  z.null(),
+  z.boolean(),
+  z.number(),
+  z.string(),
+  z.array(z.lazy(() => dataSchema)),
+  z.record(z.lazy(() => dataSchema)),
+])
```

Comment:
Oh yeah, you're totally right. I forgot about the element tree parsing incident. I can't use unknown because then the type would be different than the existing type. I'm using `z.any()` right now. Let me know if you think this is not good and we need to change it:

```ts
export type Data =
  | undefined
  | null
  | boolean
  | number
  | string
  | Data[]
  | { [key: string]: Data }

export const dataSchema: z.ZodType<Data> = z.any()
```